src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\User\Security;
+
+use Sylius\Component\User\Model\UserInterface;
+
+/**
+ * Pbkdf2PasswordEncoder uses the PBKDF2 (Password-Based Key Derivation Function 2).
+ *
+ * Providing a high level of Cryptographic security,
+ *  PBKDF2 is recommended by the National Institute of Standards and Technology (NIST).
+ *
+ * But also warrants a warning, using PBKDF2 (with a high number of iterations) slows down the process.
+ * PBKDF2 should be used with caution and care.
+ *
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ * @author Andrew Johnson
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+class UserPbkdf2PasswordEncoder implements UserPasswordEncoderInterface
+{
+    const MAX_PASSWORD_LENGTH = 4096;
+
+    /**
+     * @var string
+     */
+    private $algorithm;
+
+    /**
+     * @var bool
+     */
+    private $encodeHashAsBase64;
+
+    /**
+     * @var int
+     */
+    private $iterations;
+
+    /**
+     * @var int
+     */
+    private $length;
+
+    /**
+     * @param string $algorithm          The digest algorithm to use
+     * @param bool   $encodeHashAsBase64 Whether to base64 encode the password hash
+     * @param int    $iterations         The number of iterations to use to stretch the password hash
+     * @param int    $length             Length of derived key to create
+     */
+    public function __construct($algorithm = 'sha512', $encodeHashAsBase64 = true, $iterations = 1000, $length = 40)
```

Comment:
It is exactly the same as in Symfony. It is just a component's default class and after bump to 5.5 we can change it to use `password_hash` (http://php.net/password_hash). In bundle we use Symfony classes anyway, so it doesn't matter.